### PR TITLE
Wizard: TIMEZONE 2b - Add validation for NTP servers (HMS-5114)

### PR DESCRIPTION
**Pull request by regexowl**
This adds a validation pattern and a validation for NTP servers field and tests to check the functionality.

Based on and blocked by https://github.com/osbuild/image-builder-frontend/pull/2613

### CODE REVIEW

File: src/Components/CreateImageWizard/validators.ts [hunk 1]
```diff
@@ -81,3 +83,10 @@ export const getDuplicateMountPoints = (partitions: Partition[]): string[] => {
   }
   return duplicates;
 };
+
+export const isNtpServerValid = (ntpServer: string) => {
+  return (
+    ntpServer !== undefined &&
+    /^[a-z0-9-]+((\.[a-z0-9-]+)?){1,}$/.test(ntpServer)
```
Comment (ondrejbudai) [inline src/Components/CreateImageWizard/validators.ts:90]: what about IPv6? 😇 I wonder what format of IPv6 addresses chrony accepts.

Comment (regexowl) [reply src/Components/CreateImageWizard/validators.ts:90]: The regex can be refined for sure as this is just something I've tried to put together. Will need to do some reading on formats accepted by chrony for NTP servers thought :sweat_smile:

Comment (regexowl) [reply src/Components/CreateImageWizard/validators.ts:90]: Hmm, found something: https://chrony-project.org/doc/3.4/chrony.conf.html, let me try to update the regex a bit...

**Review by lucasgarfield — APPROVED** [review]
Works great and seems to support IPV6! Thank you!

**Comment by lucasgarfield** [comment]
pr_check failing on a timeout, admin merging.

19:35:41 2024-12-09 19:35:41 [   ERROR] [          MainThread] hit timeout error: Could not do 'wait for all deployed resources to be ready' at /var/lib/jenkins/jenkins-osbuild-image-builder-frontend-pr-check-4680-zMQyHm/.bonfire_venv/lib64/python3.8/site-packages/bonfire/openshift.py:85 in time

**Pull request merged**